lib/oxide-vpc/src/engine/gateway/mod.rs

```diff
@@ -275,3 +282,87 @@ impl Display for VpcMeta {
         write!(f, "vpc-meta")
     }
 }
+
+fn make_holepunch_rules(
+    guest_mac: MacAddr,
+    gateway_mac: MacAddr,
+    dest: IpCidr,
+    vpc_mappings: Arc<VpcMappings>,
+) -> (Rule<Finalized>, Rule<Finalized>) {
+    let vpc_meta = Arc::new(VpcMeta::new(vpc_mappings));
```

Comment:
n00b2 Q: Would it make sense to implement `MetaAction` (used with `vpc_meta` / `Action::Meta` below) on vpc_mappings directly or impl it on `Arc<VpcMappings>` so you don't have to initialize the Arc here (again)?

Reply:
The double-`Arc` is really unfortunate for sure. We could `impl MetaAction for VpcMappings`, but I think we'd need to recast the design of the `MetaAction` to use something additional like `RuleDisplay` instead of `Display`, I think?

Reply:
@FelixMcFelix ah interesting, I'll look further at this then.